sale_order_client_order_ref_autofill/models/sale_order.py

```diff
@@ -0,0 +1,17 @@
+# © 2022 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    client_order_ref = fields.Char(string="Work Number")
+    _sql_constraints = [
+        ('client_order_ref_unique', 'unique(client_order_ref)', 'Work Number must be unique!')
+    ]
+
+    def action_confirm(self):
+        self.client_order_ref = self.env['ir.sequence'].next_by_code('sale.order.work.number')
```

Comment:
When order is in draft mode, client ref could be set, and this line overwrites it. IMO it should only be right of client ref was not editable in draft mode, otherwise, previous data should be preserved. @lmiguens-solvos what do you think?

Reply:
I think it should not be editable in draft mode.
But if it was editable, then it must be overwritten with the sequence when confirm, it is important.